components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -1283,6 +1283,17 @@ void AdsServiceImpl::OnFailedToPrefetchNewTabPageAd(
                                /*intentional*/ base::DoNothing());
 }
 
+void AdsServiceImpl::ParseAndSaveCreativeNewTabPageAds(
+    const std::string& json,
+    ParseAndSaveCreativeNewTabPageAdsCallback callback) {
+  if (!bat_ads_associated_remote_.is_bound()) {
+    return std::move(callback).Run(/*success*/ false);
+  }
+
+  bat_ads_associated_remote_->ParseAndSaveCreativeNewTabPageAds(
```

Comment:
is this parsing in the browser process? if so it needs to use JSON sanitizer or similar to pass https://chromium.googlesource.com/chromium/src/+/master/docs/security/rule-of-2.md